### PR TITLE
增强instanceId兼容性，调整status同步逻辑

### LINKED ISSUE #1016: eureka服务只要求instanceId在同服务内唯一，北极星要求全局唯一
**What is the feature you want to add?**
希望北极星处理eureka client的请求时能对齐instanceId的唯一标准

**Why do you want to add this feature?**
业务方在使用eureka client时没有统一标准，出现instanceId = IP:PORT的配置，在容器环境ip复用的情况下会出问题

**How to implement this feature?**

### PR DESCRIPTION
**Pull request by MI-cool**
**Please provide issue(s) of this PR:**
Fixes #1016 

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] ApiServer

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any user-facing changes. This may include API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: apiserver/eurekaserver/write.go [hunk 1]
```diff
@@ -31,11 +32,25 @@ import (
 	"github.com/polarismesh/polaris/common/utils"
 )
 
+func checkOrBuildNewInstanceId(appId string, instanceId string) string {
+	lowerAppId := strings.ToLower(appId)
+	lowerInstanceId := strings.ToLower(instanceId)
+	if strings.Contains(lowerInstanceId, lowerAppId) {
```
Comment (andrewshan) [inline apiserver/eurekaserver/write.go:38]: 这个用前缀判断？

File: apiserver/eurekaserver/server.go [hunk 2]
```diff
@@ -248,6 +249,12 @@ func (h *EurekaServer) Initialize(ctx context.Context, option map[string]interfa
 	}
 	h.enableSelfPreservation = enableSelfPreservation
 
+	if value, ok := option[optionGenerateUniqueInstId]; ok {
+		h.generateUniqueInstId = value.(bool)
```
Comment (chuntaojun) [inline apiserver/eurekaserver/server.go:253]: h.generateUniqueInstId, _ = value.(bool)

**Review by chuntaojun — CHANGES_REQUESTED** [review]

File: apiserver/eurekaserver/access.go [hunk 3]
```diff
@@ -317,7 +317,7 @@ func checkRegisterRequest(registrationRequest *RegistrationRequest, req *restful
 		writeHeader(http.StatusBadRequest, rsp)
 		return false
 	}
-	if len(registrationRequest.Instance.InstanceId) == 0 {
+	if len(registrationRequest.Instance.InstanceId) == 0 && len(registrationRequest.Instance.HostName) == 0 {
```
Comment (chuntaojun) [inline apiserver/eurekaserver/access.go:320]: @andrewshan 这个逻辑看看会不会有兼容问题？

**Comment by chuntaojun** [comment]
修复下 golangci-lint 的问题。先执行 bash import-format.sh 格式化 import 的顺序

**Review by andrewshan — APPROVED** [review]

**Pull request merged**